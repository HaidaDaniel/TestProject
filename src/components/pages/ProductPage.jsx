/** @format */
import { useEffect } from 'react'
import { Container } from 'react-bootstrap'
import { useDispatch, useSelector } from 'react-redux'
import { useParams } from 'react-router-dom'

import { Header, ProductPageBody } from '../index'
import { fetchProductRequest } from '../../redux/ducks/product'

function ProductPage() {
  const dispatch = useDispatch()
  const { productId } = useParams()

  useEffect(() => {
    dispatch(fetchProductRequest(productId))
  }, [productId])

  const product = useSelector((state) => state.product.product)

  return (
    <Container>
      <Header></Header>
      {product && <ProductPageBody product={product}></ProductPageBody>}
    </Container>
  )
}
export default ProductPage

// ProductPage.propTypes = {
//   data: detailedProductShape.isRequired,
// }
