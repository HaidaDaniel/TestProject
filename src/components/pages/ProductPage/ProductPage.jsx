/** @format */
// import { detailedProductShape } from '../../PropTypes/DetailedProductShape/DetailedProductShape'
import Header from '../../Header/Header'
import ProductPageBody from '../../ProductPageBody/ProductPageBody'
import { Container } from 'react-bootstrap'

function ProductPage({ data }) {
  return (
    <Container>
      <Header></Header>
      {data && <ProductPageBody data={data}></ProductPageBody>}
    </Container>
  )
}
export default ProductPage

// ProductPage.propTypes = {
//   data: detailedProductShape.isRequired,
// }
