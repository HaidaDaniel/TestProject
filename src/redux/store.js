import { createStore, applyMiddleware, compose } from 'redux';
import createSagaMiddleware from 'redux-saga';

import rootReducer from './rootReducer';
import { authRootSaga, deAuthRootSaga } from './ducks/auth';
import { productRootSaga } from './ducks/product';
import { productsRootSaga } from './ducks/products';


const rootSagas = [authRootSaga, deAuthRootSaga, productsRootSaga, productRootSaga]

const sagaMiddleware = createSagaMiddleware();

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const store = createStore(
    rootReducer,
    composeEnhancers(applyMiddleware(sagaMiddleware)));

rootSagas.forEach((saga) => sagaMiddleware.run(saga));

export default store
    ;