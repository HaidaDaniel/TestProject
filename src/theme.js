const theme = {
    colors: {
        primary: '#0d6efd',
        secondary: '#6c757d',
        success: '#28a745',
        error: '#dc3545',
        golden: '#f5a623'
    },
    fonts: {
        body: 'Arial, sans-serif',
        heading: 'Helvetica, sans-serif',
    },

};

export default theme;