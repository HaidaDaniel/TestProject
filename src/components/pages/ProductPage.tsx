/** @format */
import { useEffect } from 'react'
import { Container } from 'react-bootstrap'
import { useDispatch, useSelector } from 'react-redux'
import { useParams } from 'react-router-dom'

import { Header, ProductPageBody } from '../index'
import { fetchProductRequest } from '../../redux/ducks/product'
import { RootState } from '../../redux/rootReducer'

const ProductPage = () => {
  const dispatch = useDispatch()
  const { productId } = useParams()
  useEffect(() => {
    productId && dispatch(fetchProductRequest(productId))
  }, [productId])

  const product = useSelector((state: RootState) => state.product.product)

  return (
    <Container>
      <Header />
      {product && <ProductPageBody {...product} />}
    </Container>
  )
}

export default ProductPage
