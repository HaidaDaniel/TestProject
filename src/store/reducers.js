import { combineReducers } from 'redux';

const products = (state = [], action) => {
    switch (action.type) {
        case 'FETCH_PRODUCT_SUCCESS':
            return action.payload;
        default:
            return state;
    }
};

export default combineReducers({ products });