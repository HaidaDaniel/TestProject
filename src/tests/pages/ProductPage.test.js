import '@testing-library/jest-dom/extend-expect';
import { render, screen, waitFor } from '@testing-library/react';
import { MemoryRouter, Route, Routes } from 'react-router-dom';
import { Provider } from 'react-redux';
import { ThemeProvider } from 'styled-components';

import theme from '../../theme';
import store from '../../redux/store';
import ProductPage from '../../pages/ProductPage';

describe('ProductPage Component', () => {
    it('should renders ProductPage page', async () => {
        const productId = '1';

        render(
            <Provider store={store}>
                <ThemeProvider theme={theme}>
                    <MemoryRouter initialEntries={[`/products/${productId}`]}>
                        <Routes>
                            <Route path='/products/:productId' element={<ProductPage />} /></Routes>
                    </MemoryRouter>

                </ThemeProvider>
            </Provider>
        );
        const headerElement = screen.getByText('Shop');
        expect(headerElement).toBeInTheDocument();

        await waitFor(() => {

            const productItemElement = screen.getByTestId('product-page-body');
            expect(productItemElement).toBeInTheDocument();

        });
        expect(document.body).toMatchSnapshot();

    });

});

