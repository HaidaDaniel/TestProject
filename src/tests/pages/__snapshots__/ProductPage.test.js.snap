// Jest Snapshot v1, https://goo.gl/fbAQLP

<<<<<<<< HEAD:src/tests/__snapshots__/components/pages/ProductPage.snap.js
exports[`ProductPage Component should render ProductPage page 1`] = `
========
exports[`ProductPage Component should renders ProductPage page 1`] = `
>>>>>>>> origin/newTest:src/tests/pages/__snapshots__/ProductPage.test.js.snap
<body>
  <div>
    <div
      class="container"
    >
      <nav
        class="sc-jrkPcm hELBzT navbar navbar-expand-lg navbar-light"
      >
        <a
          class="sc-kDnyCx ctmCGs mx-4"
          href="/"
        >
          Shop
        </a>
        <button
          aria-controls="basic-navbar-nav"
          aria-label="Toggle navigation"
          class="navbar-toggler collapsed"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="sc-ipMvLY dWioSl justify-content-end mx-2 navbar-collapse collapse"
          id="basic-navbar-nav"
        >
          <div
            class="ml-auto mx-2 navbar-nav"
          >
            <a
              class="nav-link"
              href="/registration"
            >
              Registration
            </a>
            <a
              class="nav-link"
              href="/login"
            >
              Login
            </a>
          </div>
        </div>
      </nav>
      <div
        class="mt-3 row"
        data-testid="product-page-body"
      >
        <div
          class="col-md-6 col-12"
        >
          <img
            alt="Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops"
            class="sc-lltjXc bwDiYp col-left"
            src="https://fakestoreapi.com/img/81fPKd-2AYL._AC_SL1500_.jpg"
            style="max-width: 100%; height: auto;"
          />
        </div>
        <div
          class="col-right mt-3 col-md-6 col-12"
        >
          <div
            class="mb-3 row"
          >
            <h3>
              Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops
            </h3>
          </div>
          <div
            class="align-items-center mb-3 row"
          >
            <div
              class="col-auto"
            >
              <div
                class="sc-hhWzdI ceIvgb"
              >
                <div
                  class="sc-bZCZoC fgPjyH"
                >
                  <span
                    class="sc-gXCJSa sc-cCIIiH jyhNbF gDBIgA"
                    data-testid="filled"
                  >
                    ★
                  </span>
                </div>
                <div
                  class="sc-bZCZoC fgPjyH"
                >
                  <span
                    class="sc-gXCJSa sc-cCIIiH jyhNbF gDBIgA"
                    data-testid="filled"
                  >
                    ★
                  </span>
                </div>
                <div
                  class="sc-bZCZoC fgPjyH"
                >
                  <span
                    class="sc-gXCJSa sc-cCIIiH jyhNbF gDBIgA"
                    data-testid="filled"
                  >
                    ★
                  </span>
                </div>
                <div
                  class="sc-bZCZoC fgPjyH"
                >
                  <span
                    class="sc-gXCJSa sc-cCIIiH jyhNbF gDBIgA"
                    data-testid="filled"
                  >
                    ★
                  </span>
                </div>
                <div
                  class="sc-bZCZoC fgPjyH"
                >
                  <span
                    class="sc-gXCJSa jyhNbF"
                    data-testid="empty"
                  >
                    ★
                  </span>
                </div>
                3.9
                 
                /5
              </div>
            </div>
            <div
              class="ml-auto my-auto col-auto"
            >
              Price:
               109.95 
              usd
            </div>
          </div>
          <div
            class="row"
          >
            <h5>
              Description:
            </h5>
          </div>
          <div
            class="mb-3 row"
          >
            Your perfect pack for everyday use and walks in the forest. Stash your laptop (up to 15 inches) in the padded sleeve, your everyday
          </div>
          <div
            class="mb-3 border-top row"
          >
            <div
              class="sc-bcPKhP dyGGkR"
            >
              <div
                class="comment-block mt-2"
                data-testid="comments-block"
              >
                <h5>
                  Customer Reviews :
                </h5>
                <div
                  class="vstack gap-3"
                >
                  <div
                    data-testid="comment"
                  >
                    <div
                      class="sc-hBpgZr iiqjsx card"
                    >
                      <div
                        class="sc-fEyylQ jaPLwZ card-header"
                      >
                        <div
                          class="sc-hhWzdI ceIvgb"
                        >
                          <div
                            class="sc-bZCZoC fgPjyH"
                          >
                            <span
                              class="sc-gXCJSa sc-cCIIiH jyhNbF gDBIgA"
                              data-testid="filled"
                            >
                              ★
                            </span>
                          </div>
                          <div
                            class="sc-bZCZoC fgPjyH"
                          >
                            <span
                              class="sc-gXCJSa sc-cCIIiH jyhNbF gDBIgA"
                              data-testid="filled"
                            >
                              ★
                            </span>
                          </div>
                          <div
                            class="sc-bZCZoC fgPjyH"
                          >
                            <span
                              class="sc-gXCJSa sc-cCIIiH jyhNbF gDBIgA"
                              data-testid="filled"
                            >
                              ★
                            </span>
                          </div>
                          <div
                            class="sc-bZCZoC fgPjyH"
                          >
                            <span
                              class="sc-gXCJSa sc-cCIIiH jyhNbF gDBIgA"
                              data-testid="filled"
                            >
                              ★
                            </span>
                          </div>
                          <div
                            class="sc-bZCZoC fgPjyH"
                          >
                            <span
                              class="sc-gXCJSa sc-cCIIiH jyhNbF gDBIgA"
                              data-testid="filled"
                            >
                              ★
                            </span>
                          </div>
                          5
                           
                          /5
                        </div>
                      </div>
                      <div
                        class="sc-fnOeiS gVDnCQ card-body"
                      >
                        <p
                          class="sc-bjEwCx dsxgdt card-text"
                        >
                          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
                        </p>
                        <p
                          class="sc-idyqAC kjTTLI card-text"
                        >
                          author:
                          Test1
                           
                          <br />
                           review id:
                          1
                        </p>
                      </div>
                    </div>
                  </div>
                  <div
                    data-testid="comment"
                  >
                    <div
                      class="sc-hBpgZr iiqjsx card"
                    >
                      <div
                        class="sc-fEyylQ jaPLwZ card-header"
                      >
                        <div
                          class="sc-hhWzdI ceIvgb"
                        >
                          <div
                            class="sc-bZCZoC fgPjyH"
                          >
                            <span
                              class="sc-gXCJSa sc-cCIIiH jyhNbF gDBIgA"
                              data-testid="filled"
                            >
                              ★
                            </span>
                          </div>
                          <div
                            class="sc-bZCZoC fgPjyH"
                          >
                            <span
                              class="sc-gXCJSa sc-cCIIiH jyhNbF gDBIgA"
                              data-testid="filled"
                            >
                              ★
                            </span>
                          </div>
                          <div
                            class="sc-bZCZoC fgPjyH"
                          >
                            <span
                              class="sc-ciJnBw gxVxox"
                              data-testid="half-filled"
                            >
                              ★
                            </span>
                          </div>
                          <div
                            class="sc-bZCZoC fgPjyH"
                          >
                            <span
                              class="sc-gXCJSa jyhNbF"
                              data-testid="empty"
                            >
                              ★
                            </span>
                          </div>
                          <div
                            class="sc-bZCZoC fgPjyH"
                          >
                            <span
                              class="sc-gXCJSa jyhNbF"
                              data-testid="empty"
                            >
                              ★
                            </span>
                          </div>
                          2.5
                           
                          /5
                        </div>
                      </div>
                      <div
                        class="sc-fnOeiS gVDnCQ card-body"
                      >
                        <p
                          class="sc-bjEwCx dsxgdt card-text"
                        >
                          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
                        </p>
                        <p
                          class="sc-idyqAC kjTTLI card-text"
                        >
                          author:
                          Test2
                           
                          <br />
                           review id:
                          2
                        </p>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="border-top pt-3 mb-3 row"
          >
            <div>
              Login please to make a review
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
